File: java/challenges/ClassifyStrings.java
import java.util.regex.Matcher;
import java.util.regex.Pattern;

public final class ClassifyStrings {

    String check(String s) {
        Pattern reg = Pattern.compile("[aeiou]{3}|[^aeiou]{5}");
        return (reg.matcher(s).find() ? "bad" : "good");
    }

    String classifyStrings(String s) {
        String res = "";
        if(!s.contains("?")) { res = check(s); }
        else {
            String s1 = check(s.replace("?", "a"));
            String s2 = check(s.replace("?", "n"));
            if(s.length() > 4 && s.contains("???")) { res = "mixed"; }
            else if(s1.equals("bad") && s2.equals("bad")) { res = "bad"; }
            else if(s1.equals("good") && s2.equals("good")) { res = "good"; }
            else { res = "mixed"; }
        }
        return res;
    }
}

File: Challenges/Ruby/classifyStrings.rb
def classifyStrings s
    g = -> (a) do
        return (/[^aeiou]{5}/ =~ a or
        /[aeiou]{3}/ =~ a) ? "bad" : "good"
    end
    res = nil
    unless s.include? "?" then res = g.call s
    else
        s1 = g.call s.gsub("?", "a")
        s2 = g.call s.gsub("?", "n")
        if s.size > 4 and s.include? "???" then res = "mixed"
        elsif s1.eql? "bad" and s2.eql? "bad" then res = "bad"
        elsif s1.eql? "good" and s2.eql? "good" then res = "good"
        else res = "mixed"
        end
    end
    return res
end